packages/gotrue/lib/src/gotrue_client.dart

```diff
@@ -132,6 +132,40 @@ class GoTrueClient {
   /// Returns the current session, if any;
   Session? get currentSession => _currentSession;
 
+  Future<AuthResponse> signInAnonymously({
+    Map<String, dynamic>? data,
+    String? captchaToken,
+  }) async {
+    try {
+      _removeSession();
+
+      final response = await _fetch.request(
+        '$_url/signup',
+        RequestMethodType.post,
+        options: GotrueRequestOptions(
+          headers: _headers,
+          body: {
+            'data': data ?? {},
+            'gotrue_meta_security': {'captcha_token': captchaToken},
+          },
+        ),
+      );
+
+      final authResponse = AuthResponse.fromJson(response);
+
+      final session = authResponse.session;
+      if (session != null) {
+        _saveSession(session);
+        notifyAllSubscribers(AuthChangeEvent.signedIn);
+      }
+
+      return authResponse;
+    } catch (error) {
+      notifyException(error);
```

Comment:
The other sign in methods don't have a try catch, which I think is better, because this method is only directly called by the user, so exceptions don't need to go to the stream.

Reply:
Makes sense!